src/internals/legend.js

```diff
@@ -513,15 +515,47 @@ extend(ChartInternal.prototype, {
 			.attr("x", $$.isLegendRight || $$.isLegendInset ? xForLegendRect : -200)
 			.attr("y", $$.isLegendRight || $$.isLegendInset ? -200 : yForLegendRect);
 
-		l.append("line")
-			.attr("class", CLASS.legendItemTile)
-			.style("stroke", $$.color)
-			.style("pointer-events", "none")
-			.attr("x1", $$.isLegendRight || $$.isLegendInset ? x1ForLegendTile : -200)
-			.attr("y1", $$.isLegendRight || $$.isLegendInset ? -200 : yForLegendTile)
-			.attr("x2", $$.isLegendRight || $$.isLegendInset ? x2ForLegendTile : -200)
-			.attr("y2", $$.isLegendRight || $$.isLegendInset ? -200 : yForLegendTile)
-			.attr("stroke-width", config.legend_item_tile_height);
+		const hasCustomPoints = $$.config.point_pattern.length;
```

Comment:
The current just is forcing to be displayed with the defined point patterns. In this case I think there're limitation of:
- When point patterns are defined, there're no option to handle legend display style
- When `point.type="rectangle"`(or 'circle') is set, they're displayed as the old square legend style

So, how about having an option like `legend.usePoint`.
```js
bb.generate({
    legend: {
        usePoint: true // default false
    }
}
```

When set this option to true, the legend will likely, displayed as data point, rather than the current square.